tests/features/payments.feature

```diff
@@ -4,30 +4,48 @@ Feature: The payments API
-        | my credit  | 1       | credit       | True       | False      | 0.0            | Jimmy Jones  | 08/2018 | Visa       | 4444333322221111 | None       | None      |
-        | my debit   | 2       | debit        | True       | False      | 0.0            | Jenny Joples | 12/2020 | Mastercard | 1111222233334444 | None       | None      |
-        | my paypal  | 1       | paypal       | False      | False      | 0.0            | Jimmy Jones  | None    | None       | None             | jj@aol.com | True      |
-        | deleted cc | 2       | credit       | False      | True       | 25.0           | Jenny Joples | 08/2018 | Visa       | 4444333322221111 | None       | None      |
-
+    Given the following "payments"
+        | nickname   | user_id | payment_type | user_name    | expires | card_type  | card_number      | user_email | is_linked | 
+        | my credit  | 1       | credit       | Jimmy Jones  | 08/2018 | Visa       | 4444333322221111 | None       | None      |
+        | my debit   | 2       | debit        | Jenny Joples | 12/2020 | Mastercard | 1111222233334444 | None       | None      |
+        | my paypal  | 1       | paypal       | Jimmy Jones  | None    | None       | None             | jj@aol.com | True      |
 
 Scenario: The server is running
     When I visit the "home page"
     Then I should see "Welcome to payments"
     Then I should not see "Not Found"
 
 Scenario: Add a new payment
-    When I visit "/payments"
+    When I visit "payments"
     Then I should see "my credit"
     Then I should see "my debit"
     Then I should see "my paypal"
-    Given a new credit card:
+    Given a new credit card
         | nickname   | user_id | payment_type | user_name   | expires | card_type | card_number      |
         | new credit | 1       | credit       | Jimmy Jones | 07/2017 | Visa      | 3333222244441111 |
-    When I add a new payment to "/payments"
-    And I visit "/payments"
+    When I add a new payment to "payments"
+    And I visit "payments"
     Then I should see "new credit"
     Then I should see "my credit"
     Then I should see "my debit"
-    Then I should see "my paypal"
\ No newline at end of file
+    Then I should see "my paypal"
+
+Scenario: Update a payment (PATCH)
+    When I get "payments" with id "1"
+    Then I should see a payment with id "1" and "nickname" = "my credit"
+    When I change "nickname" to "new credit"
+    And I patch "payments" with id "1"
+    When I get "payments" with id "1"
+    Then I should see a payment with id "1" and "nickname" = "new credit"
+
+Scenario: Update a payment (PUT)
+    When I get "payments" with id "1"
+    Then I should see a payment with id "1" and "nickname" = "my credit"
+    Given an updated credit card
+        | nickname  | user_id | payment_type | user_name   | expires | card_type | card_number      |
+        | cashmoney | 1       | credit       | Jimmy Jones | 08/2018 | Visa      | 4444333322221111 |
+    When I put "payments" with id "1"
```

Comment:
in a sense it does. the `when` in patch crates the attribute dict to be passed  as json request to PATCH. similarly, the `given` builds the new card to be passed as json request to PUT. i thought this approach might be more revealing in stating differences between PUT/PATCH, but perhaps that consideration is not so important from a client perspective